File: android/src/main/java/expo/modules/contenttransition/records/TypesetLine.kt
package expo.modules.contenttransition.records

import androidx.compose.ui.text.TextLayoutResult

internal class GlyphPlacement(
  val unit: String,
  val slot: GlyphSlot,
  val x: Float,
  val advance: Float,
  val layout: TextLayoutResult
)

internal class TypesetLine(
  val text: String,
  val units: List<String>,
  val placements: List<GlyphPlacement>,
  val width: Float,
  val height: Float,
  val baseline: Float
)

File: web/records/TypesetLine.ts
import type { GlyphSlot } from './GlyphSlot';

interface GlyphPlacement {
  readonly unit: string;
  readonly slot: GlyphSlot;
  readonly x: number;
  readonly advance: number;
}

interface TypesetLine {
  readonly placements: GlyphPlacement[];
  readonly width: number;
  readonly height: number;
}

export type { GlyphPlacement, TypesetLine };
